<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Week 2 Read Mapping | Diagnostic Genome Analysis</title>
  <meta name="description" content="Week 2 Read Mapping | Diagnostic Genome Analysis" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Week 2 Read Mapping | Diagnostic Genome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="https://github.com/mkempenaar/diagnostic-genome-analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Week 2 Read Mapping | Diagnostic Genome Analysis" />
  
  
  

<meta name="author" content="Marcel Kempenaar" />


<meta name="date" content="2024-03-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="intro-ngs-and-genetics-project.html"/>
<link rel="next" href="analysis-of-variants.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#usage"><i class="fa fa-check"></i>Usage</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html"><i class="fa fa-check"></i><b>1</b> Intro <em>NGS and Genetics</em> Project</a>
<ul>
<li class="chapter" data-level="1.1" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#cardiomyopathy"><i class="fa fa-check"></i><b>1.1</b> Cardiomyopathy</a></li>
<li class="chapter" data-level="1.2" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#medulloblastoma"><i class="fa fa-check"></i><b>1.2</b> Medulloblastoma</a></li>
<li class="chapter" data-level="1.3" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#tools"><i class="fa fa-check"></i><b>1.3</b> Tools</a></li>
<li class="chapter" data-level="1.4" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#data"><i class="fa fa-check"></i><b>1.4</b> Data</a></li>
<li class="chapter" data-level="1.5" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#analysis"><i class="fa fa-check"></i><b>1.5</b> Analysis</a></li>
<li class="chapter" data-level="1.6" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#literature"><i class="fa fa-check"></i><b>1.6</b> Literature</a></li>
<li class="chapter" data-level="1.7" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#reporting"><i class="fa fa-check"></i><b>1.7</b> Reporting</a></li>
<li class="chapter" data-level="1.8" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#assignments-week-1"><i class="fa fa-check"></i><b>1.8</b> Assignments Week 1</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#galaxy-server"><i class="fa fa-check"></i><b>1.8.1</b> Galaxy Server</a></li>
<li class="chapter" data-level="1.8.2" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#data-and-quality-control"><i class="fa fa-check"></i><b>1.8.2</b> Data and quality control</a></li>
<li class="chapter" data-level="1.8.3" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#fastq-format"><i class="fa fa-check"></i><b>1.8.3</b> FASTQ format</a></li>
<li class="chapter" data-level="1.8.4" data-path="intro-ngs-and-genetics-project.html"><a href="intro-ngs-and-genetics-project.html#quality-control"><i class="fa fa-check"></i><b>1.8.4</b> Quality Control</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="read-mapping.html"><a href="read-mapping.html"><i class="fa fa-check"></i><b>2</b> Read Mapping</a>
<ul>
<li class="chapter" data-level="2.1" data-path="read-mapping.html"><a href="read-mapping.html#mapping-with-bwa"><i class="fa fa-check"></i><b>2.1</b> Mapping with BWA</a></li>
<li class="chapter" data-level="2.2" data-path="read-mapping.html"><a href="read-mapping.html#assignment-week-2"><i class="fa fa-check"></i><b>2.2</b> Assignment week 2</a></li>
<li class="chapter" data-level="2.3" data-path="read-mapping.html"><a href="read-mapping.html#marking-duplicate-mapped-reads"><i class="fa fa-check"></i><b>2.3</b> Marking Duplicate Mapped Reads</a></li>
<li class="chapter" data-level="2.4" data-path="read-mapping.html"><a href="read-mapping.html#visualizing-the-mapping-data"><i class="fa fa-check"></i><b>2.4</b> Visualizing the Mapping Data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html"><i class="fa fa-check"></i><b>3</b> Analysis of Variants</a>
<ul>
<li class="chapter" data-level="3.1" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#assignment-week-3"><i class="fa fa-check"></i><b>3.1</b> Assignment week 3</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#programming-assignments-coverage-overview-for-panel-genes"><i class="fa fa-check"></i><b>3.1.1</b> Programming Assignments; Coverage Overview for Panel Genes</a></li>
<li class="chapter" data-level="3.1.2" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#assignment-instructions"><i class="fa fa-check"></i><b>3.1.2</b> Assignment Instructions</a></li>
<li class="chapter" data-level="3.1.3" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#assignment-1-loading-the-bed-data"><i class="fa fa-check"></i><b>3.1.3</b> Assignment 1; Loading the BED-data</a></li>
<li class="chapter" data-level="3.1.4" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#assignment-2-bed-visualization"><i class="fa fa-check"></i><b>3.1.4</b> Assignment 2; BED-visualization</a></li>
<li class="chapter" data-level="3.1.5" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#assignment-3-bioconductor"><i class="fa fa-check"></i><b>3.1.5</b> Assignment 3; Bioconductor</a></li>
<li class="chapter" data-level="3.1.6" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#assignment-4-gene-lengths"><i class="fa fa-check"></i><b>3.1.6</b> Assignment 4; Gene Lengths</a></li>
<li class="chapter" data-level="3.1.7" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#assignment-5-loading-the-mapped-reads"><i class="fa fa-check"></i><b>3.1.7</b> Assignment 5; Loading the Mapped Reads</a></li>
<li class="chapter" data-level="3.1.8" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#rsamtools"><i class="fa fa-check"></i><b>3.1.8</b> Rsamtools</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#assignment-6-pileup-processing"><i class="fa fa-check"></i><b>3.2</b> Assignment 6; Pileup Processing</a></li>
<li class="chapter" data-level="3.3" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#assignment-7-coverage-calculation-and-reporting"><i class="fa fa-check"></i><b>3.3</b> Assignment 7; Coverage calculation and reporting</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#hints"><i class="fa fa-check"></i><b>3.3.1</b> Hints</a></li>
<li class="chapter" data-level="3.3.2" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#end-result"><i class="fa fa-check"></i><b>3.3.2</b> End Result</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#finding-variants-using-lofreq"><i class="fa fa-check"></i><b>3.4</b> Finding Variants using <code>LoFreq</code></a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#vcf-file-filtering"><i class="fa fa-check"></i><b>3.4.1</b> VCF File Filtering</a></li>
<li class="chapter" data-level="3.4.2" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#assignment-8-vcf-file-processing"><i class="fa fa-check"></i><b>3.4.2</b> Assignment 8; VCF File Processing</a></li>
<li class="chapter" data-level="3.4.3" data-path="analysis-of-variants.html"><a href="analysis-of-variants.html#assignment-9-variant-visualization"><i class="fa fa-check"></i><b>3.4.3</b> Assignment 9; Variant Visualization</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="variant-annotation.html"><a href="variant-annotation.html"><i class="fa fa-check"></i><b>4</b> Variant Annotation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="variant-annotation.html"><a href="variant-annotation.html#snpeff-and-snpsift"><i class="fa fa-check"></i><b>4.1</b> SnpEff and SnpSift</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="variant-annotation.html"><a href="variant-annotation.html#assignment"><i class="fa fa-check"></i><b>4.1.1</b> Assignment</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="variant-annotation.html"><a href="variant-annotation.html#programming-assignment-analyzing-variant-annotation"><i class="fa fa-check"></i><b>4.2</b> Programming Assignment; Analyzing Variant Annotation</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="variant-annotation.html"><a href="variant-annotation.html#assignment-10-data-preparation"><i class="fa fa-check"></i><b>4.2.1</b> Assignment 10; Data Preparation</a></li>
<li class="chapter" data-level="4.2.2" data-path="variant-annotation.html"><a href="variant-annotation.html#assignment-11-finding-variants-of-interest"><i class="fa fa-check"></i><b>4.2.2</b> Assignment 11; Finding Variants of Interest</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="variant-annotation.html"><a href="variant-annotation.html#galaxy-workflow"><i class="fa fa-check"></i><b>4.3</b> Galaxy Workflow</a></li>
<li class="chapter" data-level="4.4" data-path="variant-annotation.html"><a href="variant-annotation.html#comparative-genomics"><i class="fa fa-check"></i><b>4.4</b> Comparative Genomics</a></li>
<li class="chapter" data-level="4.5" data-path="variant-annotation.html"><a href="variant-annotation.html#galaxy-data-cleanup"><i class="fa fa-check"></i><b>4.5</b> Galaxy Data Cleanup</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Diagnostic Genome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="read-mapping" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Week 2</span> Read Mapping<a href="read-mapping.html#read-mapping" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Now that we know what the quality is of the NGS patient data and have corrected the reads on low quality and short read lengths, we can continue with the mapping of our reads against a reference genome. In this step we use the human genome as a reference to which we will map the cleaned reads using the tool BWA. After mapping the reads we can look for variations between the reference and our patient.</p>
<div id="mapping-with-bwa" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Mapping with BWA<a href="read-mapping.html#mapping-with-bwa" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before starting with mapping the data to the human reference genome, we can calculate some important statistics to see what we can achieve with our data.</p>
</div>
<div id="assignment-week-2" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Assignment week 2<a href="read-mapping.html#assignment-week-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
The <em><a href="https://en.wikipedia.org/wiki/Coverage_(genetics)">coverage</a></em> is the number of reads that were mapped at a given position and it is considered a good measurement to determine if there was enough data for further analysis. To identify significant variations, we aim for a minimal coverage of <strong>20</strong>, any lower and it becomes hard to differentiate between a true variant and a sequencing error. Given the Illumina platform and given these facts:
<ul>
<li>
a minimal average coverage (read depth) of 20,
</li>
<li>
the read length is determined by your data set (for instance a read pair with each read having 75 or 150 bases),
</li>
<li>
the human genome size is 3,137,161,264 bases and
</li>
<li>
the target region (captured region, size of the gene panel) totals 320,000 bases for the Cardio panel or 1,000,000 bases for Medulloblastoma. This differs for your chosen case and has to be calculated by you.
</li>
</ul>
we can use the Lander/Waterman equation to calculate our actual expected coverage: <span class="math display">\[ C = LN / G \]</span>
Where:
<ul>
<li>
<code>C</code> is the coverage,
</li>
<li>
<code>L</code> is the read length and
</li>
<li>
<code>N</code> is the number of reads
</li>
</ul>
<p>From your own cleaned data set, look up how many reads are left for mapping. Calculate the expected coverage if we use this data for your gene panel (<code>G</code> = captured region). An Illumina MiSeq V3 can produce up to 25 million reads of length 300, how many patients could you analyse per run if the minimum coverage is 20 and you where using your gene panel? Calculate these assignments in R within your document.</p>
<p>Select the correct <strong>Map with BWA</strong> tool from the Tools menu (depends on your read length). For the mapping we will be using a built-in genome of which there are <em>many</em> available. Select the <strong>Human reference hg38</strong> in the reference genome setting. Select paired end reads.</p>
<p>Note: the reference genomes are made available by the developers of Galaxy. They host a collection of prebuilt <em>indexed</em> reference genomes that most mapping tools require. As far as we know, once someone used a certain reference genome, it is cached on our Galaxy server. If this is not the case, runtimes can become longer as several gigabytes of data need to be downloaded.</p>
<p>From your previous experiments with different FASTQ cleaning settings, select the R1 paired Trimmomatic data as the first set of reads and the R2 paired Trimmomatic as the second set of reads. Execute the tool, this can take a while to run (20 - 60 minutes).</p>
<p><img src="images/BWAMEMSettings.png" width="510" /></p>
</div>
<div id="marking-duplicate-mapped-reads" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Marking Duplicate Mapped Reads<a href="read-mapping.html#marking-duplicate-mapped-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the process of creating the reads, <em><a href="http://www.cureffi.org/2012/12/11/how-pcr-duplicates-arise-in-next-generation-sequencing/">duplicates</a></em> may have arisen by PCR artifacts. These duplicate reads are not real biological sequences in the sense that they originate from the sample, but are a technical artifact. In downstream analysis these duplicate reads may generate false positive variants. Can you think of a reason why this is the case?</p>
<p>Before we are going to look at any differences between the reference and our patient, we first have to <em>mark</em> the duplicate mapped reads. To do this, select the <a href="http://broadinstitute.github.io/picard/">MarkDuplicates</a> tool from the Tools menu. Select the <em>Map with BWA output on data … and ….</em>, set the <strong>Assume input file is already sorted option</strong> to No and Execute the tool. This tool will add a <em>flag</em> to each read that it finds as being duplicate and other tools will ignore any read that has this flag. It will therefore not <em>remove</em> the read from the data.</p>
<p><img src="images/removeDuplicates.png" width="509" /></p>
</div>
<div id="visualizing-the-mapping-data" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Visualizing the Mapping Data<a href="read-mapping.html#visualizing-the-mapping-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We are going to look at the actual mapping to get a bit of feel for what has happend till now. To do this we will look at the mapping output from the previous step - with the marked duplicates - in a <em>Genome Browser</em>.</p>
<p>On our system the <em>Integrated Genome Viewer</em> (<a href="http://software.broadinstitute.org/software/igv/">IGV</a>) has been installed. First we need to download the mapping data to our computer. To do this, download the <strong>dataset</strong> and <strong>bam_index</strong> files from the markduplicates output in Galaxy as shown below.</p>
<p><img src="images/bamDownload.png" width="510" /></p>
<p>Select <strong>Save File</strong> in the pop up window.</p>
<p>Open IGV either by going to the (Linux) <strong> Applications Menu -&gt; Run Program…</strong> and type in <strong>igv</strong> and click on <strong>launch</strong> or by opening a terminal and entering the <strong>igv</strong> command. Next, you can load the mapping data into IGV by clicking on <strong>File -&gt; load from File…</strong>. Look in your <strong>Downloads</strong> folder for a file name starting with <strong>Galaxy</strong> and ending with <strong>.bam</strong> (you only need to open the BAM file, the index file is automatically loaded).</p>
<!--
The IGV program is standard installed with version 18 of the human reference genome. In our galaxy workflow we have been using the newer 19 version. The first thing we should do is tell IGV where to find the newer genome. 
In IGV select the <strong>genomes menu</strong> and select <strong>Load Genome from File...</strong>. The <strong>Look in:</strong> will show that you are in your home directory. First go to <strong>/</strong>, than select the <strong>commons</strong> folder, next click on the <strong>minor</strong>, <strong>projectgenomics</strong> and <strong>genomes</strong> folders. From the last folder select the <strong>hg19.genome</strong> and click on <strong>Open</strong>.
-->
<p>Because our sequence reads are from captured exomes (for example the 320.000 bases for the example Cardio panel, which is just 0,001% of the total genome), you have to zoom in quite a bit to see any of the mappings. Too help you find where to zoom in, we can add an extra layer to the genome browser (called a track). Files containing all the exome regions of each genepanel are available on the galaxy server. You can download these files for your case by going to <strong>Shared Data -&gt; Data Libraries -&gt; case_name</strong> in the Galaxy browser. Select <strong>case_name.BED</strong> and click on the <strong>to History</strong> button. Please have a look at the file in you History. The file consists of 4 columns, which describe the chromosome number, start location exome, end location exome and gene name. Download this file (<em>Save File</em>) to your computer.</p>
<p>Now from IGV, again select <strong>File -&gt; Load from File..</strong>. Look in your Downloads folders for a file ending in <strong>.bed</strong> and open this file. Your screen should look like this now:</p>
<p><img src="images/igvMain.png" width="512" /></p>
<p>From the 4th column in the bed file, choose a couple of gene names. I will take <code>SOD2</code> as an example (<code>SOD2</code> lays on chr6 and has 5 exomes and is part of the Cardio gene panel).</p>
<p>In IGV type in the name of your selected gene in the search box and click on Go.</p>
<p><img src="images/IGVsearch.png" width="336" /></p>
<p>The screen will load the mapping results of the region that includes the example <code>SOD2</code> gene.
A couple of regions are important in this genome browser screen. The top row shows the location you are looking at now.</p>
<p><img src="images/IGVlocation.png" width="510" /></p>
<p>The bottom row are the locations of the reference human genes and the locations of our cardiopanel captured exomes.
<img src="images/IGVrefseqBed.png" width="512" /></p>
<p>The middle row is the actual mapping data. The first row shows a coverage plot. You can hover over the plot with the mouse. It will show how many reads where mapped at this position and what the nucloetide distribution is at this position. Also the number of forward and reverse reads is shown. In this case at this position there where 328 reads mapped. It also tells us that 100% of the reads have a G at that position (157 in the forward mapped reads and 171 in the reverse mapped reads)</p>
<p><img src="images/IGVcoverage.png" width="510" /></p>
<p>The middle row shows the mapped reads. We mapped paired-end reads and to make this visible in IGV, right click the mapping track and select <strong>View as pairs</strong>. Reads are colored according to their read orientation and insert size. Look in the <a href="http://www.broadinstitute.org/software/igv/AlignmentData">IGV online manual</a> for the explaination of the colors.</p>
<p>Zoom in on your gene of interest. Regions of reads that are grey of color indicate a simular region as the reference. Variants are shown by colored vertical bars (each nucleotide has its own color). Zoom in till you have the nucleotide sequence showing for a variant. In our example we are looking at a T variant for this patient at this position. We see that a total of 117 reads were mapped at this position and that from all reads 64 had a T and 53 had a A at this position. The patient is heterozygous for this allele. Can you see if this variant is in an exon or not?, what are the consequences of a variant in an exon location? Look for a variant in an exon. The bottom row will show the translation from DNA to protein. Does the found variant causes a change (<em>non-synonymous</em>) or is the aminoacid sequence the same (<em>synonymous</em>)?</p>
<p><img src="images/IGVvariant.png" width="114" /></p>
<p>Look at the reads at the end of an exon. The cardiopanel captured exons + 20 bases of 55 genes. Why are some reads outside of this location?</p>
<p>During the following steps we will answer the following questions <em>for each gene</em>: how many variants are found?, how many are in the exons? how many variants actually cause the aminoacid sequence to change? IGV uses a very naive variant calling method and this will result in many false positives. So we are going to use a more sophisticated method in the downstream steps.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="intro-ngs-and-genetics-project.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="analysis-of-variants.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
